dockerfiles/ci/tici/Dockerfile

```diff
@@ -20,12 +19,24 @@ ENV PATH=/root/.tiup/bin:$PATH
 RUN yum install -y python3 python3-devel mysql-devel
 RUN pip3 install protobuf==3.6.1 mysql mysql-connector-python==8.0.23
 
-# Use a patch version of CDC for output handles.
-# If the feature of pull/1403 merged, remove this command.
+# For rustup toolchain
+ENV RUSTUP_DIST_SERVER="https://mirrors.ustc.edu.cn/rust-static" RUSTUP_UPDATE_ROOT="https://mirrors.ustc.edu.cn/rust-static/rustup"
+RUN rustup install nightly-2025-02-28
+
+# Install Go and build ticdc
+RUN wget -q https://go.dev/dl/go1.23.10.linux-amd64.tar.gz && \
+    rm -rf /usr/local/go && tar -C /usr/local -xzf go1.23.10.linux-amd64.tar.gz && \
+    go version && \
+    rm go1.23.10.linux-amd64.tar.gz
+
 RUN git clone https://github.com/pingcap/ticdc.git && \
     cd ticdc && \
     git fetch origin pull/1403/head:pr-1403 && git checkout pr-1403 && \
     make cdc && \
     mv bin/cdc /usr/local/bin/cdc && \
     cd .. && \
     rm -rf ticdc
+
+# Dataset (put it at last because it takes a long time to download)
+# It could be cached elsewhere to reduce size of the image
+RUN wget -q https://quickwit-datasets-public.s3.amazonaws.com/hdfs-logs-multitenants.json.gz && gzip -d hdfs-logs-multitenants.json.gz
```

Comment:
Why is it necessary to cache this large test file here? The CI will prepare this test file in advance, and it is not recommended to include it in the image as it would increase the image size.v